scripts/plot_operator_breakdown.py

```diff
@@ -65,25 +94,70 @@
 
 df = df.fillna(0)
 
+# Obtain copy for normalized plots
+df_norm = df.copy()
+
 # Calculate share of total execution time (i.e., longer running benchmark items are weighted more)
-df.loc["Total"] = df.sum() / df.count()
+df_norm.loc["Total"] = df_norm.sum() / df_norm.count()
 
 # Normalize data from nanoseconds to percentage of total cost (calculated by dividing the cells value by the total of
 # the row it appears in)
-df.iloc[:, 0:] = df.iloc[:, 0:].apply(lambda x: x / x.sum(), axis=1)
+df_norm.iloc[:, 0:] = df_norm.iloc[:, 0:].apply(lambda x: x / x.sum(), axis=1)
 
 # Print the dataframe for easy access to the raw numbers
-print(df)
+print(df_norm)
 
 # Drop all operators that do not exceed 1% in any query
-df = df[df > 0.01].dropna(axis="columns", how="all")
+df_norm = df_norm[df_norm > 0.01].dropna(axis="columns", how="all")
+df_norm_queries = df_norm.loc[df_norm.index != "Total"]
+df_norm_total = df_norm.loc[["Total"]]
 
 # Setup colorscheme - using cubehelix, which provides a color mapping that gracefully degrades to grayscale
-colors = sns.cubehelix_palette(n_colors=len(df), rot=2, reverse=True, light=0.9, dark=0.1, hue=1)
+colors = sns.cubehelix_palette(n_colors=len(df_norm), rot=2, reverse=True, light=0.9, dark=0.1, hue=1)
 cmap = LinearSegmentedColormap.from_list("my_colormap", colors)
 
-# Plot it
-ax = df.plot.bar(stacked=True, figsize=(2 + len(df) / 4, 4), colormap=cmap)
+# Factor to adapt the sizes and ratios to large benchmarks with many queries (e.g., Join Order Benchmark).
+# The sqrt() somewhat accomodates that we don't create extremely wide charts but still adapt.
+query_count_apaption = int(round(math.sqrt(len(df_norm))))
+
+fig = plt.figure(figsize=(20 + query_count_apaption, 6))
+plt.subplots(constrained_layout=True)
+# Create a grid with two rows and ensure at least a ratio for the plots of the queries and the summary plot of
+# 4:1. If there are many queries, we increase the share of the query plots.
+gs = GridSpec(2, 5 + query_count_apaption)
+
+ax_1 = fig.add_subplot(gs[0, :-1])  # Queries, relative
+ax_2 = fig.add_subplot(gs[0, -1])  # Summary, relative
+ax_3 = fig.add_subplot(gs[1, :-1])  # Queries, absolute
+ax_4 = fig.add_subplot(gs[1, -1])  # Summary, absolute
+
+df_norm_queries.plot.bar(ax=ax_1, stacked=True, colormap=cmap)
+ax_1.yaxis.set_major_formatter(ticker.PercentFormatter(xmax=1.0))
+ax_1.set_ylabel("Share of run time\n(hiding operators <1%)")
```

Comment:
Nitpick: This is inconsistent with how it is calculated. `1%` is excluded, too.

Reply:
VLDB Errata? :D

Reply:
Filtering with `<=` now.

Reply:
I feel like `<1%` would be more common/expected than `<=1%`

Reply:
I agree, but it's worse to read on the axis title. :)

Your call.

Reply:
I'd leave the label the way it is in the master and adjust the script accordingly.

Reply:
Hmpf, that's what I meant and what I did (it's actually `>=` in the filtering). Too late ...